src/OrchardCore.Modules/OrchardCore.Roles/Services/RoleStore.cs

```diff
@@ -67,10 +67,17 @@ public async Task<IdentityResult> CreateAsync(IRole role, CancellationToken canc
                 throw new ArgumentNullException(nameof(role));
             }
 
+            var roleToCreate = (Role)role;
```

Comment:
Maybe here we should use `is` directive

Reply:
Didn't want to change the behavior but will check

Reply:
Just used the same pattern as used below for the existing `IRoleRemovedEventHandler`.

            var roleToRemove = (Role)role;